### PR TITLE
Adds sscooters and skateboards [serious pr do not close after AFD]

### PR DESCRIPTION
**Pull request by Qwertytoforty**
## What Does This PR Do

Adds skateboards. And scooters.
Skateboards come in 4 forms, makeshift, normal, pro, hoverboard (and admin)
Higher quality boards take less stamina damage from tricks / collisons, and are faster.
Skateboards are fast, especially when high quality, but will bonk off anything solid, launch people if they collide with other carbons, and are high risk high reward. They can grind on pipes, tables, and railings, to move around fast, but are more vunerable at that time.
Skateboard is 250 credits, and Pro is 600 from the crew rewards vendor.
Hoverboard is 1000 from cargo.

## Why It's Good For The Game

A fun high risk but high reward method to move across the station fast! Expensive boards are the real fast ones, but at 600 or 1000 credits, they will not be too common.
Able to do tricks, be stowed fast, and to clown on the bartender as you grind across his bar.

## Images of changes

https://youtu.be/cTiM1MJ1o7c?feature=shared

## Testing

Lots and lots of skateboarding

## Changelog
:cl:
add: Adds scooters and skateboards from TG station
/:cl:

### CODE REVIEW

**Comment by autocephalophagy** [comment]
skamtebord

**Comment by spacegeese** [comment]
the files show clown car content, god yes

File: code/__DEFINES/vehicle_defines.dm [hunk 1]
```diff
@@ -0,0 +1,55 @@
...
+// As above but only used for riding cyborgs, and only reserves 1 arm instead of 2
+#define RIDER_NEEDS_ARM (1<<1)
+/// Do we need legs to ride this (checks against TRAIT_FLOORED)
+#define RIDER_NEEDS_LEGS   (1<<2)
+/// If the rider is disabled or loses their needed limbs, do they fall off?
+#define UNBUCKLE_DISABLED_RIDER (1<<3)
+// For fireman carries, the carrying human needs an arm
+#define CARRIER_NEEDS_ARM (1<<4)
+// This rider must be our friend
+#define JUST_FRIEND_RIDERS (1<<5)
```
Comment (DGamerL) [inline code/__DEFINES/vehicle_defines.dm:30]: Please change the comments to only mention stuff we have here, so remove mentions of borg riding, firemany carry etc

File: code/__DEFINES/vehicle_defines.dm [hunk 2]
```diff
@@ -0,0 +1,55 @@
...
+
+//car_traits flags
+///Will this car kidnap people by ramming into them?
+#define CAN_KIDNAP (1<<0)
+
+#define CLOWN_CANNON_INACTIVE 0
+#define CLOWN_CANNON_BUSY 1
+#define CLOWN_CANNON_READY 2
```
Comment (DGamerL) [inline code/__DEFINES/vehicle_defines.dm:38]: We don't use these, can you remove them?

File: code/__DEFINES/vehicle_defines.dm [hunk 3]
```diff
@@ -0,0 +1,55 @@
...
+
+//Vim defines
+///cooldown between uses of the sound maker
+#define VIM_SOUND_COOLDOWN (1 SECONDS)
+///how much vim heals per weld
+#define VIM_HEAL_AMOUNT 20
```
Comment (DGamerL) [inline code/__DEFINES/vehicle_defines.dm:44]: I don't think we use these aswell

File: code/datums/components/riding/riding.dm [hunk 4]
```diff
@@ -0,0 +1,284 @@
+/**
+ * This is the riding component, which is applied to a movable atom by the [ridable element][/datum/element/ridable] when a mob is successfully buckled to said movable.
+ *
+ * This component lives for as long as at least one mob is buckled to the parent. Once all mobs are unbuckled, the component is deleted, until another mob is buckled in
+ * and we make a new riding component, so on and so forth until the sun explodes.
+ */
+
+
```
Comment (DGamerL) [inline code/datums/components/riding/riding.dm:8]:
```suggestion
```

File: code/datums/components/riding/riding.dm [hunk 5]
```diff
@@ -0,0 +1,284 @@
...
+/datum/component/riding
+	dupe_mode = COMPONENT_DUPE_UNIQUE
+
+	var/last_move_diagonal = FALSE
+	///tick delay between movements, lower = faster, higher = slower
```
Comment (DGamerL) [inline code/datums/components/riding/riding.dm:13]:
```suggestion
	/// Tick delay between movements, lower = faster, higher = slower
```

**Review by DGamerL — CHANGES_REQUESTED** [review]
Bit big of a requested change, but can you apply comments to all newly added `SIGNAL_HANDLER`s like the proposed changes for them in #24824 ?

**Comment by FunnyMan3595** [comment]
Could we get a description update now that April 1 is past?

File: code/modules/crafting/recipes.dm [hunk 6]
```diff
@@ -477,6 +477,15 @@
 	category = CAT_WEAPONRY
 	subcategory = CAT_WEAPON
 
+/datum/crafting_recipe/skateboard
+	name = "Camera Obscura"
+	result = list(/obj/tgvehicle/scooter/skateboard/improvised)
+	time = 15
```
Comment (Henri215) [inline code/modules/crafting/recipes.dm:483]:
```suggestion
	name = "Camera Obscura"
	result = list(/obj/tgvehicle/scooter/skateboard/improvised)
	time = 1.5 SECONDS
```
Name is incorrect.

File: code/modules/supply/supply_packs/pack_miscellaneous.dm [hunk 7]
```diff
@@ -525,6 +525,13 @@
 	contains = list(/obj/item/vending_refill/cargodrobe)
 	containername = "cargo clothing vendor crate"
 
+/datum/supply_packs/misc/hoverboard
+	name = "Hoverboard Crate"
+	contains = list(/obj/item/melee/skateboard/hoverboard)
+	cost = 1000 //Price of cool
+	containername = "Hoverboard Crate"
+
```
Comment (Henri215) [inline code/modules/supply/supply_packs/pack_miscellaneous.dm:533]:
```suggestion
	contains = list(/obj/item/melee/skateboard/hoverboard)
	cost = 1000 //Price of cool
	containername = "hoverboard crate"
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 8]
```diff
@@ -0,0 +1,342 @@
...
+			return
+		if((istype(bumped_thing, /obj/machinery/economy/vending)))
+			var/obj/machinery/economy/vending/V = bumped_thing
+			rider.Weaken(8 SECONDS)
+			visible_message("<span class='danger'>[src] crashes into [V]!</span>")
+			V.tilt(rider, from_combat = TRUE)
+			return
+		rider.throw_at(throw_target, 3, 2)
+		var/head_slot = rider.get_item_by_slot(SLOT_HUD_HEAD)
+		if(!head_slot || !(istype(head_slot,/obj/item/clothing/head/helmet) || istype(head_slot,/obj/item/clothing/head/hardhat)))
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:116]:
```suggestion
		if(!head_slot || !(istype(head_slot ,/obj/item/clothing/head/helmet) || istype(head_slot, /obj/item/clothing/head/hardhat)))
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 9]
```diff
@@ -0,0 +1,342 @@
...
+	var/piping_hot = FALSE
+	for(var/obj/machinery/atmospherics/pipe/P in loc.contents)
+		if(P.invisibility == 0 && (loc.layer == 1.7 || P.layer == 2.47))
+			piping_hot = TRUE
+			break
+	if(!(locate(/obj/structure/table) in loc.contents) && !(locate(/obj/structure/railing) in loc.contents) && !piping_hot)
+		grinding = FALSE
+		icon_state = "[initial(icon_state)]"
+		return
+	skater.adjustStaminaLoss(instability*0.3)
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:156]:
```suggestion
	skater.adjustStaminaLoss(instability * 0.3)
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 10]
```diff
@@ -0,0 +1,342 @@
...
+		skater.Weaken(4 SECONDS)
+		grinding = FALSE
+		icon_state = "[initial(icon_state)]"
+		return
+	playsound(src, 'sound/effects/skateboard_roll.ogg', 50, TRUE)
+	var/turf/location = get_turf(src)
+
+	if(location)
+		if(prob(33))
+			location.hotspot_expose(1000,1000)
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:172]:
```suggestion
			location.hotspot_expose(1000, 1000)
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 11]
```diff
@@ -0,0 +1,342 @@
...
+	if(skater.incapacitated() || !Adjacent(skater))
+		return
+	if(has_buckled_mobs())
+		to_chat(skater, "<span class='warning'>You can't lift this up when somebody's on it.</span>")
+		return
+	skater.put_in_hands(new board_item_type(get_turf(skater)))
+	qdel(src)
+
+/obj/tgvehicle/scooter/skateboard/pro
+	name = "skateboard"
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:201]:
```suggestion
	name = "pro skateboard"
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 12]
```diff
@@ -0,0 +1,342 @@
...
+	desc = "An EightO brand professional skateboard. Looks a lot more stable than the average board."
+	icon_state = "skateboard2"
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:203]: Better rename it to "skateboard_pro" or something other than just 2.

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 13]
```diff
@@ -0,0 +1,342 @@
...
+	icon_state = "scooter_frame"
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/scooter_frame/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/stack/sheet/metal))
+		return ..()
+	if(!I.tool_start_check(user, amount=5))
+		return
+	to_chat(user, "<span class='notice'>You begin to add wheels to [src].</span>")
+	if(!I.use_tool(src, user, 80, volume=50, amount=5))
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:249]:
```suggestion
	if(!I.use_tool(src, user, 80, volume = 50, amount = 5))
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 14]
```diff
@@ -0,0 +1,342 @@
...
+	name = "scooter frame"
+	desc = "A metal frame for building a scooter. Looks like you'll need to add some iron to make wheels."
+	icon = 'icons/obj/tgvehicles.dmi'
+	icon_state = "scooter_frame"
+	w_class = WEIGHT_CLASS_NORMAL
+
+/obj/item/scooter_frame/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/stack/sheet/metal))
+		return ..()
+	if(!I.tool_start_check(user, amount=5))
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:246]:
```suggestion
	if(!I.tool_start_check(user, amount = 5))
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 15]
```diff
@@ -0,0 +1,342 @@
...
+/obj/tgvehicle/scooter/skateboard/wrench_act(mob/living/user, obj/item/I)
+	return
+
+/obj/tgvehicle/scooter/skateboard/improvised/attackby(obj/item/I, mob/user, params)
+	if(!istype(I, /obj/item/stack/rods))
+		return ..()
+	if(!I.tool_start_check(user, amount=2))
+		return
+	to_chat(user, "<span class='notice'>You begin making handlebars for [src].</span>")
+	if(!I.use_tool(src, user, 25, volume=50, amount=2))
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:272]:
```suggestion
	if(!I.tool_start_check(user, amount = 2))
		return
	to_chat(user, "<span class='notice'>You begin making handlebars for [src].</span>")
	if(!I.use_tool(src, user, 25, volume = 50, amount = 2))
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 16]
```diff
@@ -0,0 +1,342 @@
...
+		unbuckle_mob(skaterboy)
+		skaterskoot.buckle_mob(skaterboy)
+	qdel(src)
+
+/obj/tgvehicle/scooter/skateboard/improvised/screwdriver_act(mob/living/user, obj/item/I)
+	. = ..()
+	if(.)
+		return
+	to_chat(user, "<span class='notice'>You begin to deconstruct and remove the wheels on [src]...</span>")
+	if(!I.use_tool(src, user, 20, volume=50))
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:287]:
```suggestion
	if(!I.use_tool(src, user, 20, volume = 50))
```

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 17]
```diff
@@ -0,0 +1,342 @@
...
+	S.buckle_mob(user)
+	qdel(src)
+
+/obj/item/melee/skateboard/improvised
+	name = "improvised skateboard"
+	desc = "A jury-rigged skateboard. It can be placed on its wheels and ridden, or used as a radical weapon."
+	board_item_type = /obj/tgvehicle/scooter/skateboard/improvised
+
+/obj/item/melee/skateboard/pro
+	name = "skateboard"
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/scooter.dm:324]:
```suggestion
	name = "pro skateboard"
```

File: code/modules/vehicle/tg_vehicles/tg_vehicle_actions.dm [hunk 18]
```diff
@@ -0,0 +1,285 @@
...
+/datum/action/vehicle/scooter/skateboard/ollie/Trigger(left_click)
+	. = ..()
+	if(!.)
+		return
+	var/obj/tgvehicle/scooter/skateboard/vehicle = vehicle_target
+	if(vehicle.grinding)
+		return
+	var/mob/living/rider = owner
+	var/turf/landing_turf = get_step(vehicle.loc, vehicle.dir)
+	rider.adjustStaminaLoss(vehicle.instability* 0.75)
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/tg_vehicle_actions.dm:216]:
```suggestion
	rider.adjustStaminaLoss(vehicle.instability * 0.75)
```

File: code/modules/vehicle/tg_vehicles/tg_vehicle_actions.dm [hunk 19]
```diff
@@ -0,0 +1,285 @@
...
+	var/mob/living/rider = owner
+
+	rider.adjustStaminaLoss(board.instability)
+	if(rider.getStaminaLoss() >= 100)
+		playsound(src, 'sound/effects/bang.ogg', 20, vary = TRUE)
+		board.unbuckle_mob(rider)
+		rider.Weaken(5 SECONDS)
+		if(prob(15))
+			rider.visible_message(
+				"<span class='danger'>[rider] misses the landing and falls on [rider.p_their()] face!)</span>",
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/tg_vehicle_actions.dm:264]:
```suggestion
				"<span class='danger'>[rider] misses the landing and falls on [rider.p_their()] face!</span>",
```

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 20]
```diff
@@ -0,0 +1,244 @@
...
+		inserted_key.forceMove(drop_location())
+	inserted_key = I
+
+/obj/tgvehicle/AltClick(mob/user)
+	if(!inserted_key)
+		return ..()
+	if(!is_occupant(user))
+		to_chat(user, "<span class='warning'>You must be riding the [src] to remove [src]'s key!</span>")
+		return
+	to_chat(user, "<span class='notice'>You remove \the [inserted_key] from \the [src].</span>")
```
Comment (Henri215) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:226]: Bunch of \the in multiple places.

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 21]
```diff
@@ -0,0 +1,244 @@
...
+	pass_flags = PASSTABLE
+	COOLDOWN_DECLARE(message_cooldown)
+	COOLDOWN_DECLARE(cooldown_vehicle_move)
+	var/list/mob/occupants //mob = bitflags of their control level.
+	///Maximum amount of passengers plus drivers
+	var/max_occupants = 1
+	////Maximum amount of drivers
+	var/max_drivers = 1
+	var/movedelay = 2
+	var/lastmove = 0
```
Comment (Burzah) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:23]: Can we snake_case these?

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 22]
```diff
@@ -0,0 +1,244 @@
...
+	/**
+	  * If the driver needs a certain item in hand (or inserted, for vehicles) to drive this. For vehicles, this must be duplicated on their riding component subtype
+	  * [/datum/component/riding/var/keytype] variable because only a few specific checks are handled here with this var, and the majority of it is on the riding component
+	  * Eventually the remaining checks should be moved to the component and this var removed.
+	  */
+	var/key_type
+	///The inserted key, needed on some vehicles to start the engine
+	var/obj/item/key/inserted_key
+	/// Whether the vehicle is currently able to move
+	var/canmove = TRUE
```
Comment (Burzah) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:33]: This one too.

File: code/__DEFINES/vehicle_defines.dm (hunk 2 above)
Comment (Qwertytoforty) [reply code/__DEFINES/vehicle_defines.dm:38]: removed

File: code/__DEFINES/vehicle_defines.dm (hunk 3 above)
Comment (Qwertytoforty) [reply code/__DEFINES/vehicle_defines.dm:44]: removed

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm (hunk 20 above)
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:226]: removed

File: code/datums/components/riding/riding.dm [hunk 23]
```diff
@@ -0,0 +1,283 @@
...
+	if(isnull(.)) //you can set it to null to not change it.
+		. = AM.layer
+	AM.layer = .
+
+/datum/component/riding/proc/set_vehicle_dir_layer(dir, layer)
+	directional_vehicle_layers["[dir]"] = layer
+
+/// This is called after the ridden atom is successfully moved and is used to handle icon stuff
+/datum/component/riding/proc/vehicle_moved(datum/source, oldloc, dir, forced)
+	SIGNAL_HANDLER
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/datums/components/riding/riding.dm:128]: yes, will do

File: code/datums/components/riding/riding_vehicle.dm [hunk 24]
```diff
@@ -0,0 +1,179 @@
...
+		if(COOLDOWN_FINISHED(src, message_cooldown))
+			to_chat(user, "<span class='warning'>You cannot operate \the [vehicle_parent] right now!</span>")
+			COOLDOWN_START(src, message_cooldown, 5 SECONDS)
+		return COMPONENT_DRIVER_BLOCK_MOVE
+
+	if(ride_check_flags & RIDER_NEEDS_LEGS && HAS_TRAIT(user, TRAIT_FLOORED))
+		if(ride_check_flags & UNBUCKLE_DISABLED_RIDER)
+			INVOKE_ASYNC(vehicle_parent, TYPE_PROC_REF(/atom/movable, unbuckle_mob), user, TRUE)
+			user.visible_message("<span class='danger'>[user] falls off \the [vehicle_parent].</span>",\
+			"<span class='danger'>You fall off \the [vehicle_parent] while trying to operate it while unable to stand!</span>")
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/datums/components/riding/riding_vehicle.dm:39]: removed

File: code/datums/elements/ridable.dm [hunk 25]
```diff
@@ -0,0 +1,189 @@
...
+/datum/element/ridable/proc/check_mounting_async_edition(atom/movable/target_movable, mob/living/potential_rider, force = FALSE, ride_check_flags = NONE)
+	var/arms_needed = 0
+	if(ride_check_flags & RIDER_NEEDS_ARMS)
+		arms_needed = 2
+	else if(ride_check_flags & RIDER_NEEDS_ARM)
+		arms_needed = 1
+		ride_check_flags &= ~RIDER_NEEDS_ARM
+		ride_check_flags |= RIDER_NEEDS_ARMS
+
+	if(arms_needed && !equip_buckle_inhands(potential_rider, arms_needed, target_movable)) // can be either 1 (cyborg riding) or 2 (human piggybacking) hands
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/datums/elements/ridable.dm:55]: removed

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm (hunk 21 above)
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:23]: unused, removed

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm (hunk 22 above)
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:33]: done

File: code/modules/vehicle/tg_vehicles/scooter.dm (hunk 8 above)
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/scooter.dm]: dpme

File: code/modules/vehicle/tg_vehicles/scooter.dm (hunk 12 above)
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/scooter.dm]: done

File: code/game/objects/structures/tables_racks.dm [hunk 26]
```diff
@@ -459,7 +459,7 @@
 		check_break(M)
 
 /obj/structure/table/glass/proc/check_break(mob/living/M)
-	if(has_gravity(M) && M.mob_size > MOB_SIZE_SMALL)
+	if(has_gravity(M) && M.mob_size > MOB_SIZE_SMALL && !istype(M.buckled, /obj/tgvehicle/scooter/skateboard/hoverboard))
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/game/objects/structures/tables_racks.dm:462]: I can't imagine too many more items that would get this, I can if you think it is worthwhile however

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 27]
```diff
@@ -0,0 +1,342 @@
...
+	var/turf/location = get_turf(src)
+
+	if(location)
+		if(prob(33))
+			location.hotspot_expose(1000, 1000)
+			sparks.start() //the most radical way to start plasma fires
+	for(var/mob/living/carbon/victim in location)
+		if(victim.body_position == LYING_DOWN)
+			playsound(location, 'sound/items/trayhit2.ogg', 40)
+			victim.apply_damage(damage = 25, damagetype = BRUTE, def_zone = list("head", "chest", "l_arm", "r_arm", "l_leg", "r_leg"))
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/scooter.dm:177]: we have a define for all of them? thank god

[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/scooter.dm:177]: oh wait you want me to use defines for all of these, got it.

File: code/datums/components/riding/riding.dm [hunk 28]
```diff
@@ -0,0 +1,282 @@
...
+/datum/component/riding/proc/handle_vehicle_offsets(dir)
+	var/atom/movable/AM = parent
+	var/AM_dir = "[dir]"
+	var/passindex = 0
+	if(!AM.has_buckled_mobs())
+		return
+
+	for(var/m in AM.buckled_mobs)
+		passindex++
+		var/mob/living/buckled_mob = m
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/datums/components/riding/riding.dm:163]: no clue, typecasting

File: code/datums/components/riding/riding.dm [hunk 29]
```diff
@@ -0,0 +1,282 @@
...
+	directional_vehicle_layers["[dir]"] = layer
+
+/// This is called after the ridden atom is successfully moved and is used to handle icon stuff
+/datum/component/riding/proc/vehicle_moved(datum/source, oldloc, dir, forced)
+	SIGNAL_HANDLER //COMSIG_MOVABLE_MOVED
+
+	var/atom/movable/movable_parent = parent
+	if(isnull(dir))
+		dir = movable_parent.dir
+	for(var/m in movable_parent.buckled_mobs)
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/datums/components/riding/riding.dm:133]: doing

File: code/game/atoms_movable.dm [hunk 30]
```diff
@@ -474,6 +477,12 @@
 
 	return TRUE
 
+/atom/movable/proc/set_glide_size(target = 8)
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/game/atoms_movable.dm:480]: done

File: code/modules/vehicle/tg_vehicles/scooter.dm [hunk 31]
```diff
@@ -0,0 +1,342 @@
...
+	var/old_pass = skater.pass_flags //Re-do this, so railings don't fuck with the grinder
+	var/old_v_pass = pass_flags
+	skater.pass_flags |= PASSTABLE | PASSFENCE
+	pass_flags |= PASSTABLE | PASSFENCE
+	step(src, dir)
+	skater.pass_flags = old_pass
+	pass_flags = old_v_pass
+	var/piping_hot = FALSE
+	for(var/obj/machinery/atmospherics/pipe/P in loc.contents)
+		if(P.invisibility == 0 && (loc.layer == 1.7 || P.layer == 2.47))
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/scooter.dm:149]: yes, correcting

File: code/game/objects/structures/tables_racks.dm (hunk 26 above)
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/game/objects/structures/tables_racks.dm:462]: done

File: code/game/turfs/simulated/floor/chasm.dm [hunk 32]
```diff
@@ -130,6 +131,8 @@
 		var/mob/living/M = AM
 		if(M.flying || M.floating)
 			return FALSE
+		if(istype(M.buckled, /obj/tgvehicle/scooter/skateboard/hoverboard))
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/game/turfs/simulated/floor/chasm.dm:134]: done

**Comment by BAGELMENSK** [comment]
AUGH YEAH SOMETHIGN TO DO!!!!!!! I'm ready for every assistant on the station to be cruisin around getting arrested by sec for smashing into people

File: code/datums/elements/ridable.dm [hunk 33]
```diff
@@ -0,0 +1,181 @@
...
+
+/obj/item/riding_offhand/dropped()
+	selfdeleting = TRUE
+	. = ..()
+
+/obj/item/riding_offhand/equipped()
+	if(loc != rider && loc != parent)
+		selfdeleting = TRUE
+		qdel(src)
+	. = ..()
```
Comment (Sirryan2002) [inline code/datums/elements/ridable.dm:173]:
```suggestion
	return ..()
```

File: code/datums/elements/ridable.dm [hunk 34]
```diff
@@ -0,0 +1,181 @@
...
+	w_class = WEIGHT_CLASS_HUGE
+	flags = ABSTRACT | DROPDEL | NOBLUDGEON
+	resistance_flags = INDESTRUCTIBLE | LAVA_PROOF | FIRE_PROOF | UNACIDABLE | ACID_PROOF
+	var/mob/living/carbon/rider
+	var/mob/living/parent
+	var/selfdeleting = FALSE
+
+/obj/item/riding_offhand/dropped()
+	selfdeleting = TRUE
+	. = ..()
```
Comment (Sirryan2002) [inline code/datums/elements/ridable.dm:167]:
```suggestion
	return ..()
```

File: code/datums/elements/ridable.dm [hunk 35]
```diff
@@ -0,0 +1,181 @@
...
+		selfdeleting = TRUE
+		qdel(src)
+	. = ..()
+
+/obj/item/riding_offhand/Destroy()
+	var/atom/movable/AM = parent
+	if(selfdeleting)
+		if(rider in AM.buckled_mobs)
+			AM.unbuckle_mob(rider)
+	. = ..()
```
Comment (Sirryan2002) [inline code/datums/elements/ridable.dm:180]:
```suggestion
	return ..()
```

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 36]
```diff
@@ -0,0 +1,242 @@
...
+/obj/tgvehicle/examine(mob/user)
+	. = ..()
+	. += generate_integrity_message()
+
+/// Returns a readable string of the vehicle's health for examining. Overridden by subtypes who want to be more verbose with their health messages.
+/obj/tgvehicle/proc/generate_integrity_message()
+	var/examine_text = ""
+	var/integrity = obj_integrity/max_integrity * 100
+	switch(integrity)
+		if(50 to 99)
```
Comment (lewcc) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:66]:
```suggestion
		if(50 to INFINITY)
```

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 37]
```diff
@@ -0,0 +1,242 @@
...
+	return ..()
+
+/obj/tgvehicle/examine(mob/user)
+	. = ..()
+	. += generate_integrity_message()
+
+/// Returns a readable string of the vehicle's health for examining. Overridden by subtypes who want to be more verbose with their health messages.
+/obj/tgvehicle/proc/generate_integrity_message()
+	var/examine_text = ""
+	var/integrity = obj_integrity/max_integrity * 100
```
Comment (lewcc) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:64]:
```suggestion
	var/integrity = obj_integrity / max_integrity * 100
```

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 38]
```diff
@@ -0,0 +1,242 @@
...
+		if(0 to 25)
+			examine_text = "<span class='warning'>It's falling apart!</span>"
+
+	return examine_text
+
+/obj/tgvehicle/proc/is_key(obj/item/I)
+	return istype(I, key_type)
+
+/obj/tgvehicle/proc/return_occupants()
+	return occupants
```
Comment (lewcc) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:79]: Is this going to cause trouble with it being a lazylist

**Review by lewcc — COMMENTED** [review]
looks pretty rad to me 😎

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm (hunk 36 above)
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:66]: this will make it look damaged even if it is not damaged

Comment (lewcc) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:66]: oh you're so right

**Review by lewcc — APPROVED** [review]
LGTM, excited to see these in

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm (hunk 38 above)
Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:79]: uh, not sure, ask another reviewer in that regards, lists vs lazylists isn't my point of knowledge

**Review by Burzah — APPROVED** [review]

**Review by Sirryan2002 — APPROVED** [review]

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 39]
```diff
@@ -0,0 +1,242 @@
+/obj/tgvehicle
+	name = "generic vehicle"
+	desc = "Yell at coderbus."
```
Comment (DGamerL) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:3]:
```suggestion
	desc = "Yell at coding chat."
```

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm (hunk 38 above)
Comment (DGamerL) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:79]: Honestly not a clue, you might have to check if the list is initialized first / check it in the place that uses `return occupants()`
Or this, to be sure:
```suggestion
	return length(occupants) ? occupants : list()
```

Comment (Qwertytoforty) [reply code/modules/vehicle/tg_vehicles/tg_vehicles.dm:79]: replaced

**Review by DGamerL — COMMENTED** [review]
`/obj/tgvehicle` is my new nightmare

File: code/modules/mining/abandonedcrates.dm [hunk 40]
```diff
@@ -121,7 +115,7 @@
 		if(91)
 			new /obj/item/organ/internal/brain/xeno(src)
 		if(92)
-			new /obj/item/organ/internal/heart(src)
+			new /obj/item/melee/skateboard/pro(src)
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/mining/abandonedcrates.dm:118]: probably supposed to be the hoverboard!

File: code/modules/mob/inventory_procs.dm [hunk 41]
```diff
@@ -32,6 +32,11 @@
 /mob/proc/is_holding(obj/item/I)
 	return istype(I) && (I == r_hand || I == l_hand)
 
+//Checks if we're holding an item of type: typepath
+/mob/proc/is_holding_item_of_type(typepath)
+	. = FALSE
+	if(istype(l_hand, typepath) || istype(r_hand, typepath))
+		return I
```
[the remark this answers is not in the document]
Comment (Qwertytoforty) [reply code/modules/mob/inventory_procs.dm]: done locally

File: code/modules/vehicle/tg_vehicles/tg_vehicles.dm [hunk 42]
```diff
@@ -0,0 +1,242 @@
...
+/obj/tgvehicle/Move(newloc, dir)
+	// It is unfortunate, but this is the way to make it not mess up
+	var/atom/old_loc = loc
+	// When we do this, it will set the loc to the new loc
+	. = ..()
+	if(trailer && .)
+		var/dir_to_move = get_dir(trailer.loc, old_loc)
+		step(trailer, dir_to_move)
+
+
```
Comment (DGamerL) [inline code/modules/vehicle/tg_vehicles/tg_vehicles.dm:188]:
```suggestion
```

**Review by DGamerL — APPROVED** [review]

**Pull request merged**